comment/migrations/0008_comment_urlhash.py

```diff
@@ -0,0 +1,49 @@
+from django.db import migrations, models
+
+from comment.managers import CommentManager
+from comment.utils.utils import id_generator
+
+def generate_urlhash():
+    manager_obj = CommentManager()
+    return id_generator(
+        prefix=manager_obj._urlhash_prefix,
+        len_id=manager_obj._len_urlhash_id,
+        suffix=manager_obj._urlhash_suffix
+        )
+
+
+def set_unique_urlhash(Comment, obj, instance):
+    if not instance.urlhash:
+        hash_val = generate_urlhash()
+        instance.urlhash = hash_val
+        while Comment.objects.filter(urlhash=hash_val).exists():
+            instance.urlhash = generate_urlhash()
+
+
+def set_default_urlhash(apps, schema_editor):
+    Comment = apps.get_model('comment', 'Comment')
+    comment_obj = Comment()
+    for comment in Comment.objects.all():
+        set_unique_urlhash(Comment, comment_obj, comment)
+        comment.save(update_fields=['urlhash'])
+
```

Comment:
For compatibility with sonarlint, please user ``comment_model`` variable instead of ``Comment``
```suggestion
def set_unique_urlhash(comment_model, obj, instance):
    if not instance.urlhash:
        hash_val = generate_urlhash()
        instance.urlhash = hash_val
        while comment_model.objects.filter(urlhash=hash_val).exists():
            instance.urlhash = generate_urlhash()


def set_default_urlhash(apps, schema_editor):
    comment_model = apps.get_model('comment', 'Comment')
    comment_obj = comment_model()
    for comment in comment_model.objects.all():
        set_unique_urlhash(comment_model, comment_obj, comment)
        comment.save(update_fields=['urlhash'])

```

Reply:
I will change the variable name but since I currently don't use `SonarLint` I really don't understand the exact issue here.

Reply:
Me neither but sonarLint complains 😄 

<img width="918" alt="Screenshot 2020-07-02 at 13 32 02" src="https://user-images.githubusercontent.com/25413382/86353947-6e37e900-bc68-11ea-8a5c-4c1c938ef916.png">

Reply:
This is from the official django documentation. It can be found under the heading https://docs.djangoproject.com/en/3.0/howto/writing-migrations/#migrations-that-add-unique-fields
```python
# Generated by Django A.B on YYYY-MM-DD HH:MM
from django.db import migrations
import uuid

def gen_uuid(apps, schema_editor):
    MyModel = apps.get_model('myapp', 'MyModel')
    for row in MyModel.objects.all():
        row.uuid = uuid.uuid4()
        row.save(update_fields=['uuid'])

class Migration(migrations.Migration):

    dependencies = [
        ('myapp', '0004_add_uuid_field'),
    ]

    operations = [
        # omit reverse_code=... if you don't want the migration to be reversible.
        migrations.RunPython(gen_uuid, reverse_code=migrations.RunPython.noop),
    ]
```

Reply:
From what I see in the image, the rule is for normal variables and not Classes as such.

Reply:
Yes please. Simply django code recommendation is not compatible with sonarLint

Reply:
Okay, if you consider SonarLint's recommendation to be above Django's, I will change it.